backend/Generator/main.py

```
@@ -373,6 +391,18 @@ def process_file(self, file):
             content = self.extract_text_from_docx(file_path)
+        elif file.filename.endswith('.pptx'):
+            content = self.extract_text_from_pptx(file_path)
+            if not content:
+                content = ""
+        elif file.filename.endswith('.ppt'):
+            try:
+                content = self.extract_text_from_pptx(file_path)
+                if not content:
+                    with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
+                        content = f.read()
+            except Exception:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)